src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts

```diff
@@ -85,29 +86,38 @@ export class AdminHomePageComponent {
    * Adds the instructor at the i-th index.
    */
   addInstructor(i: number): void {
+    this.asyncAddInstructor(i)
+      ?.subscribe( { complete: () => this.isAddingInstructors = false, } );
+  }
+
+  /**
+   * Adds the instructor at the i-th index.
+   */
+  asyncAddInstructor(i: number): Observable<JoinLink> | null {
     const instructor: InstructorData = this.instructorsConsolidated[i];
     if (instructor.status !== 'PENDING' && instructor.status !== 'FAIL') {
```

Comment:
Lets make the status strings (PENDING, FAIL, ...) into enums

Reply:
Yup, sounds good